posthog/session_recordings/session_recording_api.py

```diff
@@ -663,6 +663,72 @@ def destroy(self, request: request.Request, *args: Any, **kwargs: Any) -> Respon
 
         return Response({"success": True}, status=204)
 
+    @extend_schema(exclude=True)
+    @action(methods=["POST"], detail=False, url_path="bulk_delete")
+    def bulk_delete(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
+        """Bulk soft delete recordings by providing a list of recording IDs."""
+
+        session_recording_ids = request.data.get("session_recording_ids", [])
+
+        if not session_recording_ids or not isinstance(session_recording_ids, list):
+            raise exceptions.ValidationError("session_recording_ids must be provided as a non-empty array")
+
+        if len(session_recording_ids) > 20:
+            raise exceptions.ValidationError("Cannot process more than 20 recordings at once")
+
+        # Load recordings from ClickHouse to get distinct_ids for ones that don't exist in Postgres
+        query = RecordingsQuery(session_ids=session_recording_ids)
+        recordings, _, _ = list_recordings_from_query(query, cast(User, request.user), self.team)
+
+        # Filter out recordings that are already deleted
+        non_deleted_recordings = [recording for recording in recordings if not recording.deleted]
+
+        # First, bulk create any missing records
+        session_recordings_to_create = [
+            SessionRecording(
+                team=self.team,
+                session_id=recording.session_id,
+                distinct_id=recording.distinct_id,
+                deleted=True,
+            )
+            for recording in non_deleted_recordings
```

Comment:
i think that because we're ignoring conflicts we can create a record without filtering the list
no idea if it makes any difference though